BDArmory/Control/BDModuleOrbitalAI.cs

```diff
@@ -651,7 +735,7 @@ void UpdateStatus()
             // Update status mode
             if (weaponManager && weaponManager.missileIsIncoming && weaponManager.incomingMissileVessel && weaponManager.incomingMissileTime <= weaponManager.evadeThreshold) // Needs to start evading an incoming missile.
                 currentStatusMode = StatusMode.Evading;
-            else if (CheckOrbitUnsafe() || belowSafeAlt)
+            else if (fixOrbitNow)
```

Comment:
The `hasPropulsion` check needs to be further up in this logic. In my latest test, several vessels were "Correcting Orbit" despite not having any fuel.

I also had one craft that had fuel, but no EC, so it was unable to point in the appropriate direction to start thrusting (it was stuck on "Intercepting Target"). I think in such circumstances, setting `fc.alignmentToleranceforBurn` higher (maybe 45° or higher?) would help as long as burning the engines produces EC.

For the "Stranded" + "Correcting Orbit" combo, you could have custom statuses such as "Stranded - Escaping" or "Stranded - Lithobraking", for example.